### PR TITLE
[#343] Refactor DeviceDetailsPage to use Page

### LINKED ISSUE #343: Refactor Device Details to use Page component

### PR DESCRIPTION
**Pull request by jun-loke**
# General info

**Issue number**: Closes #343 

**Task description**: 

 - Move DeviceDetails Page to DeviceDetails Contents
 - Create DeviceDetailsPage with Page component
 - Make the link use the Serial number (unique identifier)
 - Move DeviceNameDetail into the ConciseTableRow component
 - Extract EdtiableName and StaticName from the class
 - test all new components
 
 
| Original   | Updated|
|:---------------:|:---------------:|
| | |

# Additional notes

**Note to reviewers**:

### CODE REVIEW

File: frontend/src/deviceDetailsPage/__tests__/EditableName.test.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      />
+    );
+  });
+  describe("returns a component that", () => {
+    it("Contains 1 <Form/> component with expected props", () => {
+      expect(wrapper.find(".deviceNameEditForm")).toHaveLength(1);
+      const formProps = wrapper.find(".deviceNameEditForm").first().props();
+      expect(formProps.onSubmit).toBe(mockConfirm);
+    });
+    it("Contains 1 <Box/> component with expected props", () => {
```
Comment (gdzooks) [inline frontend/src/deviceDetailsPage/__tests__/EditableName.test.jsx:34]: no props to expect LOL

File: frontend/src/deviceDetailsPage/__tests__/DeviceName.test.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  });
+
+  describe("confirmEditing() function", () => {
+    beforeEach(() => {
+      wrapper.setState(editingState);
+    });
+
+    it("call the DeviceApi.updateDeviceName function", () => {
+      // expect(DeviceApi.updateDeviceName).toBeCalledTimes(1);
+    });
```
Comment (gdzooks) [inline frontend/src/deviceDetailsPage/__tests__/DeviceName.test.jsx]: is this meant to be filled in in the future?

**Review by gdzooks — CHANGES_REQUESTED** [review]
looking very sexy! would like my 2nd comment addressed before i can approve :pray:

**Review by felixlapierre — DISMISSED** [review]

File: frontend/src/deviceDetailsPage/__tests__/DeviceName.test.jsx (hunk 2 above)
Comment (jun-loke) [reply frontend/src/deviceDetailsPage/__tests__/DeviceName.test.jsx]: @felixlapierre was supposed to write that test

**Review by gdzooks — DISMISSED** [review]

**Review by felixlapierre — APPROVED** [review]

**Review by gdzooks — APPROVED** [review]

**Pull request merged**